files/69-available-updates-begin.sh.amzn2

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/bash
+
+# Copyright (C) 2024 Amazon.com, Inc. or its affiliates.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#    http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the
+# License.
+
+package_updates=$(LANG=C timeout 30s /usr/bin/yum \
+  --debuglevel 2 \
+  --security check-update 2>/dev/null \
+  | grep -P '(?<! 0 packages) available$')
+
+if [ -n "$package_updates" ]; then
+  echo "$package_updates
+Update to the latest ECS-Optimized AMI for all updates, including security"
+
+  security_updates=$(grep -oP '(\d+)(?=.* security)' <<< "$package_updates")
+
+  if [ ${security_updates:=0} -ne 0 ]; then
+    echo 'Run "sudo yum update --security" to apply security updates in place'
+  fi
+
+  echo 'Run "sudo yum update" to apply all updates in place'
+fi
+
+# The order in which update-motd runs the scripts is
+# 1. 69-available-updates-begin
+# 2. 70-available-updates
```

Comment:
Curious how is the order being enforced here?

Reply:
Yeah, script execution order is ["basically alphabetical order, with a few caveats"](https://manpages.ubuntu.com/manpages/bionic/man5/update-motd.5.html#description).